extide/gradle/netbeans-gradle-tooling/src/main/java/org/netbeans/modules/gradle/tooling/NetBeansRunSinglePlugin.java

```diff
@@ -43,37 +43,49 @@ class NetBeansRunSinglePlugin implements Plugin<Project> {
 
     private static final String RUN_SINGLE_TASK = "runSingle";
     private static final String RUN_SINGLE_MAIN = "runClassName";
-    private static final String RUN_SINGLE_ARGS = "runArgs";
-    private static final String RUN_SINGLE_JVM_ARGS = "runJvmArgs";
-    private static final String RUN_SINGLE_CWD = "runWorkingDir";
+    private static final String RUN_ARGS = "runArgs";
+    private static final String RUN_JVM_ARGS = "runJvmArgs";
+    private static final String RUN_JVM_DEBUG_ARGS = "runJvmDebugArgs";
+    private static final String RUN_CWD = "runWorkingDir";
 
     @Override
     public void apply(Project project) {
         project.afterEvaluate(p -> {
-            if (project.getPlugins().hasPlugin("java") 
-                    && (project.getTasks().findByPath(RUN_SINGLE_TASK) == null)
-                    && project.hasProperty(RUN_SINGLE_MAIN)) {
-                Set<Task> runTasks = p.getTasksByName("run", false);
-                Task r = runTasks.isEmpty() ? null : runTasks.iterator().next();
-                String mainClass = project.property(RUN_SINGLE_MAIN).toString();
+            if (project.getPlugins().hasPlugin("java")) {
+                String mainClass = project.hasProperty(RUN_SINGLE_MAIN) ? project.property(RUN_SINGLE_MAIN).toString()
+                                                                        : null;
                 p.getTasks().withType(JavaExec.class).configureEach(je -> {
```

Comment:
Hm .. on a second thought, maybe there should be some trigger that causes this code to decorate exec tasks; just some arbitrary property present in run/debug actions just to trigger this. Otherwise, it might reconfigure execs even for `gradle compile` ...